addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-connection-service.js

```diff
@@ -220,6 +229,90 @@ class EnvironmentScConnectionService extends Service {
     return _.get(result, 'connection') || connection;
   }
 
+  async createConnectionSSMUrl(requestContext, envId, connectionId) {
+    const [environmentScService, pluginRegistryService] = await this.service([
+      'environmentScService',
+      'pluginRegistryService',
+    ]);
+
+    const connection = await this.mustFindConnection(requestContext, envId, connectionId);
+
+    // Write audit event
+    await this.audit(requestContext, { action: 'ssm-url-requested', body: { id: envId, connection } });
+
+    const { outputs } = await environmentScService.mustFind(requestContext, { id: envId });
+
+    const [aws] = await this.service(['aws']);
+
+    const { WorkspaceSSMRoleArn, Ec2WorkspaceInstanceId, SessionDuration } = cfnOutputsArrayToObject(outputs);
+
+    const { accessKeyId, secretAccessKey, sessionToken } = await aws.getCredentialsForRole({
+      roleArn: WorkspaceSSMRoleArn,
+    });
+
+    connection.url = await this.generateSigninSSMConsoleLink(
+      accessKeyId,
+      secretAccessKey,
+      sessionToken,
+      Ec2WorkspaceInstanceId,
+      SessionDuration,
+    );
+
+    // This is done so that plugins know it was called during create URL cycle
+    connection.operation = 'create';
+    // Give plugins chance to adjust the connection (such as connection url etc)
+    const result = await pluginRegistryService.visitPlugins(
+      'env-sc-connection-url',
+      'createConnectionUrl',
+      {
+        payload: {
+          envId,
+          connection,
+        },
+      },
+      { requestContext, container: this.container },
+    );
+
+    return _.get(result, 'connection') || connection;
+  }
+
+  async generateSigninSSMConsoleLink(accessKeyId, secretAccessKey, sessionToken, instanceId, inputSessionDuration) {
+    const session = {
+      sessionId: accessKeyId,
+      sessionKey: secretAccessKey,
+      sessionToken,
+    };
+
+    let sessionDuration = 3599; // default session duration, 59:59
+    if (Number(inputSessionDuration) > 0) {
+      sessionDuration = Number(inputSessionDuration);
+    }
+    const params = {
+      Action: 'getSigninToken',
+      SessionDuration: sessionDuration,
+      Session: session,
+    };
+
+    const region = this.settings.get(settingKeys.awsRegion);
+
+    let consoleDomainSuffix;
+    if (region.startsWith('cn-')) {
+      consoleDomainSuffix = 'amazonaws.cn';
+    } else {
+      consoleDomainSuffix = 'aws.amazon.com';
+    }
+
+    const getSigninTokenURL = `https://${region}.signin.${consoleDomainSuffix}/federation`;
+
+    const response = await axios.get(getSigninTokenURL, { params });
+
+    const signinToken = response.data.SigninToken;
+
+    const ssmConsoleUrl = `https://${region}.signin.${consoleDomainSuffix}/federation?Action=login&Issuer=Instruqt&Destination=https://${region}.console.${consoleDomainSuffix}/systems-manager/session-manager/${instanceId}?region=${region}&SigninToken=${signinToken}`;
```

Comment:
the fixed issuer `Instruqt` is required? or any meaning?

Reply:
Yes, Issuer is not mandatory. 
I removed it.